oak_attestation_verification/src/verifier.rs

```diff
@@ -311,17 +315,67 @@ fn verify_cb(
     _endorsements: &CbEndorsements,
     _reference_values: &CbReferenceValues,
 ) -> anyhow::Result<()> {
-    anyhow::bail!("Needs implementation")
+    anyhow::bail!("needs implementation")
+}
+
+/// Verifies the AMD attestation report.
+fn verify_amd_attestation_report(
+    report: &[u8],
+    reference_values: &AmdSevReferenceValues,
+) -> anyhow::Result<()> {
+    let parsed = AttestationReport::ref_from(report)
+        .ok_or_else(|| anyhow::anyhow!("could not parse AMD attestation report"))?;
+    parsed.validate().map_err(|msg| anyhow::anyhow!(msg))?;
+    let data = &parsed.data;
+
+    // Verify that we are not in debug mode.
+    if !reference_values.allow_debug {
+        let policy_flags = data
+            .policy
+            .get_flags()
+            .ok_or_else(|| anyhow::anyhow!("failed to parse flags"))?;
+        if policy_flags.bits() & PolicyFlags::DEBUG.bits() != 0 {
+            anyhow::bail!("debug mode not allowed");
+        }
+    }
+
+    Ok(())
+}
+
+/// Verifies the Intel attestation report.
+fn verify_intel_attestation_report(
+    _report: &[u8],
+    _reference_values: &IntelTdxReferenceValues,
+) -> anyhow::Result<()> {
+    anyhow::bail!("needs implementation")
 }
 
 /// Verifies all ingredients of the root layer, which is common to both
 /// Oak Restricted Kernel and Oak Containers setups.
 fn verify_root_layer(
     now_utc_millis: i64,
-    _l: &RootLayerEvidence,
+    l: &RootLayerEvidence,
     e: &RootLayerEndorsements,
     r: &RootLayerReferenceValues,
 ) -> anyhow::Result<()> {
+    match l.platform() {
+        TeePlatform::Unspecified => anyhow::bail!("unspecified TEE platform"),
+        TeePlatform::AmdSevSnp => {
+            let amd_sev = r
```

Comment:
I would expect the checks to be the other way round: first check what the reference value expects, then confirm that what's inside the evidence lines up with that. As it's written, it feels a bit backwards to me. for instance what if the reference value says to skip the validation?

Reply:
Got it, makes sense indeed if there is something about the root layer that we always want to check no matter what (regardless of the reference values), thanks for clarifying.